mrmustard/lab_dev/transformations/base.py

```diff
@@ -36,6 +40,25 @@ class Transformation(CircuitComponent):
     Base class for all transformations.
     """
 
+    def inverse(self) -> Transformation:
+        r"""Returns the inverse of the transformation."""
+        if not isinstance(self.representation, Bargmann):
+            raise NotImplementedError("Only Bargmann representation is supported.")
+        if self.representation.ansatz.batch_size > 1:
+            raise NotImplementedError("Batched transformations are not supported.")
+        A, b, _ = self.dual.representation.conj().triple  # apply X
+        almost_inverse = self.__class__.from_bargmann(
+            [0], (math.inv(A[0]), -math.inv(A[0]) @ b[0], 1 + 0j)
```

Comment:
Can we somehow know the modes of this Transformation directly rather than use [0]?

Reply:
[0] is the first and only element of the batch